### PR TITLE
feat(lightsail): support lightsail Instance construct 

### PR DESCRIPTION
**Pull request by pahud**
support the `Instance` construct for Lightsail.

Fixes: #16852 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-lightsail/lib/instance.ts [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+}
+
+/**
+ * The Bundle for the instance.
+ */
+export class Bundle {
+  /**
+   * nano_2_0
+   */
+  public static readonly NANO_2_0 = Blueprint.of('nano_2_0');
```
Comment (corymhall) [inline packages/@aws-cdk/aws-lightsail/lib/instance.ts:159]: I think these should be `Bundle.of()`?

File: packages/@aws-cdk/aws-lightsail/lib/instance.ts [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+/**
+ * The properties of the instance.
+ */
+export interface InstanceProps extends cdk.ResourceProps {
+  /**
+   * The blueprint of the instance.
+   *
+   * @default AMAZON_LINUX_2
+   */
+  readonly blueprint?: Blueprint;
```
Comment (corymhall) [inline packages/@aws-cdk/aws-lightsail/lib/instance.ts:236]: I think this should be a required property. I think the user should have to think about what type of Instance they want.

**Review by corymhall — COMMENTED** [review]
@pahud thanks for the PR! I've left some initial comments, but I also think there is some additional functionality that we should include initially.

At a minimum I think the L2 needs to include:
- Configuring snapshots
- Specifying a custom key pair
- Specifying User Data
- Configuring the networking 

I think the networking is especially critical since Lightsail will create the instance open to the world by default.

Other things that I think can come later (since they depend on the creation of other L2s) are:

- configuring disks

**Comment by TheRealAmazonKendra** [comment]
This PR has been deemed to be abandoned, and will be closed. Please create a new PR for these changes if you think this decision has been made in error.

**Pull request closed without merging**